python/chatting_sample.py

```diff
@@ -0,0 +1,392 @@
+            if cancellation_details.reason == speechsdk.CancellationReason.Error:
+                print("Error details: {}".format(cancellation_details.error_details))
+
+    def pronunciation_assessment():
+        # Setup the audio stream
+        framerate, bits_per_sample, num_channels = read_wave_header(input_files[0])
+        format = speechsdk.audio.AudioStreamFormat(
+            samples_per_second=framerate,
+            bits_per_sample=bits_per_sample,
+            channels=num_channels
+        )
+        stream = speechsdk.audio.PushAudioInputStream(format)
+        audio_config = speechsdk.audio.AudioConfig(stream=stream)
+
+        speech_config = speechsdk.SpeechConfig(subscription=speech_key, region=service_region)
+
+        # Create pronunciation assessment config, set grading system,
+        # granularity and if enable miscue based on your requirement.
+        pronunciation_config = speechsdk.PronunciationAssessmentConfig(
+            grading_system=speechsdk.PronunciationAssessmentGradingSystem.HundredMark,
+            granularity=speechsdk.PronunciationAssessmentGranularity.Phoneme,
+            enable_miscue=False)
+        pronunciation_config.enable_prosody_assessment()
+        pronunciation_config.enable_content_assessment_with_topic(topic)
+
+        # Creates a speech recognizer using a file as audio input.
+        language = 'en-US'
+        speech_recognizer = speechsdk.SpeechRecognizer(
+            speech_config=speech_config,
+            language=language,
+            audio_config=audio_config
+        )
+        # Apply pronunciation assessment config to speech recognizer
+        pronunciation_config.apply_to(speech_recognizer)
+
+        done = False
+        recognized_words = []
+        prosody_scores = []
+        fluency_scores = []
+        durations = []
+        results = []
+        json_words = []
+        mis_pronunciation_words = []
+        display_text = ""
+
+        def stop_cb(evt):
+            """callback that signals to stop continuous recognition upon receiving an event `evt`"""
+            nonlocal done
+            done = True
+
+        def recognized(evt):
+            pronunciation_result = speechsdk.PronunciationAssessmentResult(evt.result)
+            nonlocal recognized_words, prosody_scores, fluency_scores, durations, results, json_words
+            nonlocal display_text, mis_pronunciation_words
+            results.append(pronunciation_result)
+            recognized_words += pronunciation_result.words
+            fluency_scores.append(pronunciation_result.fluency_score)
+            json_result = evt.result.properties.get(speechsdk.PropertyId.SpeechServiceResponse_JsonResult)
+            jo = json.loads(json_result)
+            nb = jo["NBest"][0]
+            display_text += nb["Display"]
+            json_words += nb["Words"]
+            if "ProsodyScore" in nb["PronunciationAssessment"]:
```

Comment:
Please do not get prosody score from json.

Reply:
updated